song-server/src/main/java/bio/overture/song/server/auth/AuthZAuthorizationService.java

```diff
@@ -0,0 +1,46 @@
+package bio.overture.song.server.auth;
+
+import java.util.List;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.oauth2.core.OAuth2AuthenticatedPrincipal;
+import org.springframework.stereotype.Service;
+
+@Service
+@Profile("secure")
+public class AuthZAuthorizationService {
+
+  @Value("${authz.admin.group}")
+  private String ADMIN_GROUP;
+
+  private OAuth2AuthenticatedPrincipal getPrincipalFromAuth(Authentication authentication) {
+    if (authentication.getPrincipal() instanceof OAuth2AuthenticatedPrincipal) {
+      return (OAuth2AuthenticatedPrincipal) authentication.getPrincipal();
+    }
+    return null;
+  }
+
+  public boolean isAdmin(Authentication authentication) {
```

Comment:
Should this function take an `AuthZClaims` as input instead of generic `Authentication`?